extensions/wikia/RTE/RTE.class.php

```diff
@@ -657,22 +651,22 @@ static private function getMessages() {
 	 * @return boolean true/false if we are in fancy edit mode
 	 */
 
-	static function isWysiwygModeEnabled() {
-		return (self::$useWysiwyg && self::$initMode == "wysiwyg");
+	static public function isWysiwygModeEnabled(): bool {
+		return ( self::$useWysiwyg && self::$initMode == self::INIT_MODE_WYSIWYG );
 	}
 
 	/**
 	 * This will be false if mediawiki editor is being used
 	 * @return boolean true/false if RTE is enabled
 	 */
-	static function isEnabled() {
+	static function isEnabled(): bool {
 		return self::$useWysiwyg;
 	}
 
 	/**
 	 * Add "Enable Rich Text Editing" as the first option in editing tab of user preferences
 	 */
-	static function onEditingPreferencesBefore($user, &$preferences) {
+	static function onEditingPreferencesBefore( $user, &$preferences ): bool  {
 		// add JS to hide certain switches when wysiwyg is enabled
 		global $wgOut, $wgJsMimeType, $wgExtensionsPath;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)